VideoLocker/src/main/java/org/edx/mobile/module/db/impl/IDatabaseImpl.java

```diff
@@ -110,157 +117,236 @@ public Integer updateVideoAsOnlineByVideoId(String videoId, final DataCallback<I
+                        + DbStructure.Column.USERNAME + "=?",
+                new String[]{String.valueOf(DownloadedState.DOWNLOADING.ordinal()),
+                        enrollmentId, chapter, section, username()}, null);
+        op.setCallback(callback);
+        return enqueue(op);
+    }
 
     @Override
     public Boolean isVideoDownloadedInSection(String enrollmentId,
-                                              String chapter, String section, final DataCallback<Boolean> callback) {
-        DbOperationExists op = new DbOperationExists(false,DbStructure.Table.DOWNLOADS,
-            new String[] {DbStructure.Column.VIDEO_ID},
-            DbStructure.Column.SECTION + "=? AND " + DbStructure.Column.CHAPTER + "=? AND " + DbStructure.Column.EID + "=? AND "
-                + DbStructure.Column.DOWNLOADED + "=? AND " + DbStructure.Column.USERNAME + "=?",
-            new String[] { section, chapter, enrollmentId,
-                String.valueOf(DownloadedState.DOWNLOADED.ordinal()),username()}, null);
+                                              String chapter, String section, final
+                                              DataCallback<Boolean> callback) {
+        DbOperationExists op = new DbOperationExists(false, DbStructure.Table.DOWNLOADS,
+                new String[]{DbStructure.Column.VIDEO_ID},
+                DbStructure.Column.SECTION + "=? AND " + DbStructure.Column.CHAPTER + "=? AND " +
+                        DbStructure.Column.EID + "=? AND "
+                        + DbStructure.Column.DOWNLOADED + "=? AND " + DbStructure.Column.USERNAME
+                        + "=?",
+                new String[]{section, chapter, enrollmentId,
+                        String.valueOf(DownloadedState.DOWNLOADED.ordinal()), username()}, null);
+        op.setCallback(callback);
+        return enqueue(op);
+    }
+
+    @Override
+    public long[] getDownloadedVideoDmIdsForSection(String enrollmentId, String chapter,
+                                                    String section, final
+                                                    DataCallback<List<Long>> callback) {
+        DbOperationGetColumn<Long> op = new DbOperationGetColumn<Long>(true, DbStructure.Table
+                .DOWNLOADS,
+                new String[]{DbStructure.Column.DM_ID},
+                DbStructure.Column.DOWNLOADED + "=? AND " + DbStructure.Column.EID + "=? AND "
+                        + DbStructure.Column.CHAPTER + "=? AND "
+                        + DbStructure.Column.SECTION + "=? AND "
+                        + DbStructure.Column.USERNAME + "=?",
+                new String[]{String.valueOf(DownloadedState.DOWNLOADED.ordinal()),
+                        enrollmentId, chapter, section, username()}, null, Long.class);
+        op.setCallback(callback);
+        List<Long> downloadedList = enqueue(op);
+        if (downloadedList == null) {
+            return new long[]{};
```

Comment:
this test `DbTests#testgetDownloadingVideoDmIdsForSection` fails, if this check isn't applied.